scripts/python/assets-automation/generate_assets_report.py

```diff
@@ -0,0 +1,535 @@
+    )
+
+    print("done.")
+
+    return result
+
+
+def evaluate_cpp_package(package_path: str) -> int:
+    evaluation = 0
+
+    possible_assets_json = os.path.join(package_path, "..", "assets.json")
+
+    if False:
+        evaluation = 1
+
+    if os.path.exists(possible_assets_json):
+        evaluation = 2
+
+    return evaluation
+
+
+def generate_cpp_report() -> ScanResult:
+    language = "CPP"
+    result = ScanResult(language)
+    repo_root = get_repo(language)
+
+    print(f"Evaluating repo for {language} @ {repo_root}", end="...")
+
+    exclusions = [os.path.join("vcpkg", "vcpkg.json"), "template"]
+
+    packages = glob.glob(os.path.join(repo_root, "sdk", "**", "vcpkg.json"), recursive=True)
+    packages = [os.path.dirname(pkg) for pkg in packages if not any([x in pkg for x in exclusions])]
+
+    result.packages = sorted([os.path.basename(pkg) for pkg in packages])
+    for pkg in packages:
+        evaluation = evaluate_cpp_package(pkg)
+
+        if evaluation == 1:
+            result.packages_using_proxy.append(os.path.basename(pkg))
+        elif evaluation == 2:
+            result.packages_using_proxy.append(os.path.basename(pkg))
+            result.packages_using_external.append(os.path.basename(pkg))
+
+    print("done.")
+    return result
+
+
+def evaluate_js_package(package_path: str) -> int:
+    with open(package_path, "r", encoding="utf-8") as f:
+        package_json = json.load(f)
+
+    assets_json = os.path.join(os.path.dirname(package_path), "assets.json")
+    if os.path.exists(assets_json):
+        return 2
+
+    if "devDependencies" in package_json:
+        if "@azure-tools/test-recorder" in package_json["devDependencies"]:
+            version_spec = package_json["devDependencies"]["@azure-tools/test-recorder"]
+            if version_spec[0] == "^":
+                version_spec = version_spec[1:]
+
+            if version.parse(version_spec) >= version.parse("2.0.0"):
+                return 1
+
+    return 0
+
+
+def e_startswith(input: str, prefixes: List[str]) -> bool:
+    return any([input.startswith(fix) for fix in prefixes])
+
+
+def e_endswith(input: str, postfixes: List[str]) -> bool:
+    return any([input.endswith(fix) for fix in postfixes])
+
+
+def e_directory_in(input_dir: str, directory_patterns: List[str]) -> bool:
+    return any([subdir in input_dir for subdir in directory_patterns])
+
+
+def js_package_included(package_path: str) -> bool:
+    package_name = os.path.basename(os.path.dirname(package_path))
+
+    excluded_packages = [
+        "samples-react",
+        "sample-react",
+        "mock-hub", "abort-controller",
+        "logger",
+        "samples-express",
+        "samples-browser", "samples-react",
+        "event-hubs-track-1",
+        "opentelemetry-instrumentation-azure-sdk",
+        "monitor-opentelemetry-exporter",
+        "service-bus-v1",
+        "service-bus-v7",
+        "app",
+        "perf",
```

Comment:
The ones that depend on AMQP, are to be skipped
	- service-bus
	- eventhubs-checkpointstore-blob
	- eventhubs-checkpointstore-tables
	- schema-registry-avro	

Nothing to record, so to be skipped
	- api-management-custom-widgets-scaffolder
	- storage-internal-avro
        - web-pubsub-express

Reply:
```suggestion
        "perf",
        "service-bus",
        "eventhubs-checkpointstore-blob",
        "eventhubs-checkpointstore-tables",
        "schema-registry-avro",
        "api-management-custom-widgets-scaffolder",
        "storage-internal-avro",
        "web-pubsub-express",
```